view/frontend/web/js/applepay/button.js

```diff
@@ -476,40 +479,86 @@ define([
 
                 let componentData = self.applePayComponent.data;
 
-                setShippingInformation(payload, this.isProductView).done(function () {
+                setShippingInformation(payload, this.isProductView).done(function() {
                     // Submit payment information
                     const postData = {
                         email: shippingContact.emailAddress,
                         paymentMethod: {
                             method: 'adyen_hpp',
                             additional_data: {
                                 brand_code: 'applepay',
-                                stateData: JSON.stringify(componentData)
-                            }
-                        }
+                                stateData: JSON.stringify(componentData),
+                            },
+                        },
                     };
 
-                    if (window.checkout && window.checkout.agreementIds) {
-                        postData.paymentMethod.extension_attributes = {
-                            agreement_ids: window.checkout.agreementIds
-                        };
-                    }
+                    postData.paymentMethod = agreementsAssigner(postData.paymentMethod);
```

Comment:
Hello @dimitriBouteille,

Thank you for your contribution. I see that you've implemented the usage of `agreementsAssigner()`. However, I couldn't find the relevant import in RequireJS import block. 

Is `Magento_CheckoutAgreements/js/model/agreements-assigner` required to be imported to the class?

Best regards,
Can

Reply:
Hi @candemiralp 

I think this line can be remove :)

Reply:
Hello @candemiralp 

Any news for this PR ?